### DISCUSSION TITLE
Wazuh agent causes field type conflicts in Kibana

### DISCUSSION THREAD

**Discussion by sanctanox** [Security-Onion-Solutions/securityonion#3825] [Unsupported Versions] [answered]
Clean standalone install from SO 2.3.40 ISO
Soon as I install the Wazuh agent on a windows DC (obtained from the downloads section of the SOC), field conflicts are created in the so-* index pattern.
Previous to installing the windows agent there are indexes named in the pattern of "so-ossec-YYYY.MM.DD" such as:

so-ossec-2021.04.08
so-ossec-2021.04.09
etc...

After installing the wazuh windows agent onto a windows DC, a new index is created on the standalone Security Onion server:

so-ossec

This new "so-ossec" index uses field mappings that are different from the previous indexes name "so-ossec-<YYYY.MM.DD>".  For instance:

agent.ip is a "text" field in the index named "so-ossec", but it is an "ip" field in the indices named "so-ossec-<YYYY.MM.DD>"
event.timestamp is a "date" field in the index named "so-ossec", but it is a "text" field in the indices named "so-ossec-<YYYY.MM.DD>
ingest.timestamp is a "date" field in the index named "so-ossec", but it is a "text" field in the indices named "so-ossec-<YYYY.MM.DD>"
winlog.event_id is a "text" field in the index named "so-ossec", but it is a "long" (integer) field in the indices named "so-ossec-<YYYY.MM.DD>"
winlog.version is a "text" field in the index named "so-ossec", but it is a "long" (integer) field in the indices named "so-ossec-<YYYY.MM.DD>"

Can someone point me in the right direction to adjust these field mappings?  Everything I've tried seems to get overwritten.  I don't understand why simply sending data in from a WIndows host creates a whole new index.  I would think it would just be put into the current so-ossec-YY.MM.DD index.

**Comment by sanctanox** [comment]
I think this might be the same issue or similar as in the following post:
https://github.com/Security-Onion-Solutions/securityonion/discussions/3816

**Comment by defensivedepth** [comment, marked as answer]
This is happening because of a bug - Windows Eventlogs shipped by Wazuh are being mis-indexed into an index named `so-ossec` - You can temporarily work around it by doing the following (it will be fixed in the next release):

Remove the following two lines in `/opt/so/saltstack/default/salt/elasticsearch/files/ingest/win.eventlogs`
 (lines 7 & 8)

 `{ "rename":         { "if": "ctx.winlog?.systemTime != null", "field": "@timestamp",      "target_field": "ingest.timestamp",     "ignore_missing": true } },`
  `{ "set":           { "if": "ctx.winlog?.systemTime != null",   "field": "@timestamp", "value": "{{winlog.systemTime}}", "override": true }  },`
    

Restart Elasticsearch:

`sudo so-elasticsearch-restart --force`

And then delete the `so-ossec` index:

`curl -X DELETE -k "https://localhost:9200/so-ossec?pretty"`

**Reply by sanctanox** [reply]
Thank you!  I searched for known issues/bugs, but couldn't relate this to any.  (probably didn't use the correct search terms) 

The work around has resolved the issue!

Thanks again!

**Reply by defensivedepth** [reply]
No worries, we recently uncovered this bug.
